db/range_del_aggregator.cc

```diff
@@ -50,6 +58,14 @@ bool RangeDelAggregator::ShouldDelete(const ParsedInternalKey& parsed) {
     return false;
   }
   const auto& tombstone_map = GetTombstoneMap(parsed.sequence);
+  if (collapse_deletions_) {
+    auto iter = tombstone_map.upper_bound(parsed.user_key.ToString());
```

Comment:
1. yes, not sure what I was thinking
2. lower_bound gives us the first point >= search key, but we want the last point <= search key.